src/algorithms/queueObject.js

```diff
@@ -0,0 +1,24 @@
+export default class Queue {
+  constructor(...args) {
+    this.length = args.length;
+    for(let i = 1; i <= args.length; i++) {
+      this[i] = args[i - 1];
+    }
+  }
+  push(item) {
+    this.length += 1;
+    this[this.length] = item;
+  }
+  pop() {
+    if (!this.length) {
+      throw new Error('queue is empty');
+    }
+    const firstItem = this[1];
+    for(let i = 1; i < this.length; i++) {
+      Object.defineProperty(this, i, {value: this[i +1]});
```

Comment:
you shouldn't need defineProperty here. Obviously you'll want to run the tests to ensure it works, but I don't see why you can't just do 
```suggestion
      this[i] = this[i + 1]
```

Reply:
huh, not sure why I didn't think that would work